### ISSUE TITLE
Dijkstra failing for Le Siffleur à ventre roux

### ISSUE THREAD

**Issue by Luke-Poeppel** [Luke-Poeppel/decitala#123]
Replicate with:
```
search.path_finder(
	filepath=...,
	part_num=0,
	table=GreekFootHashTable(),
)
```
Gets the following: 
```
Le_Siffleur_à_ventre_roux_Ex98.xml
Traceback (most recent call last):
  File "/Users/lukepoeppel/moiseaux/tests/test_database.py", line 15, in <module>
    make_database("/Users/lukepoeppel/moiseaux/databases/ODNC.db")
  File "/Users/lukepoeppel/moiseaux/moiseaux/database.py", line 202, in make_database
    res = search.path_finder(
  File "/Users/lukepoeppel/decitala/decitala/search.py", line 200, in path_finder
    best_path = dijkstra.generate_path(
  File "/Users/lukepoeppel/decitala/decitala/path_finding/dijkstra.py", line 68, in generate_path
    key = pred[path[0]]
KeyError: 13
```

**Comment by Luke-Poeppel** [comment]
This error was actually related to the current `best_source_and_sink` helper function (from `decitala.path_finding.path_finding_utils`). Fixing this will give me the opportunity to implement the "right" approach of comparative path ranking (w.r.t. the cost function).

**Comment by Luke-Poeppel** [comment]
Fixed in bfbab0a38dcd776c351781618b9c28d071cc6531.
